spec/controllers/api/v1/topics_controller_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::TopicsController, type: :controller do
+  describe 'with user logged in' do
+    let!(:topics)       { create_list(:topic, 4) }
+    login_user
+    
+    it "should have a current_user" do
+      expect(subject.current_user).to_not eq(nil)
+    end
+
+    it "returns a success response" do
+      get 'index'
+      payload = JSON.parse(response.body)
+      expect(payload).to_not be_empty
+      expect(response).to be_successful
+    end
+
+    it "should return all topics" do
+      get 'index'
+      payload = JSON.parse(response.body)
```

Comment:
Since you are doing this several times (and you will do it later in other files) I suggest you to define a method in `spec/helpers.rb`.

```
def json
  JSON.parse(response.body).with_indifferent_access
end

```
And here just do 
```suggestion
     get 'index'
      expect(json.size).to eq(topics.size)
```

Reply:
**Hi, I made this change and returns the following error:**

Api::V1::TopicsController with user logged in returns a successfull response
     Failure/Error: JSON.parse(response.body).with_indifferent_access
     
     NoMethodError:
       undefined method `with_indifferent_access' for [{"id"=>257, "name"=>"Thad Kulas III", "image"=>"http://thiel.net/inge_gorczany", "created_at"=>"2023-05-30T22:32:33.649Z", "updated_at"=>"2023-05-30T22:32:33.649Z"}, {"id"=>258, "name"=>"Dave Lynch", "image"=>"http://stamm.co/delphine_becker", "created_at"=>"2023-05-30T22:32:33.656Z", "updated_at"=>"2023-05-30T22:32:33.656Z"}, {"id"=>259, "name"=>"Kam Labadie", "image"=>"http://yost-smith.co/gregory", "created_at"=>"2023-05-30T22:32:33.664Z", "updated_at"=>"2023-05-30T22:32:33.664Z"}, {"id"=>260, "name"=>"Ian Miller", "image"=>"http://torphy-kilback.co/rima_daniel", "created_at"=>"2023-05-30T22:32:33.670Z", "updated_at"=>"2023-05-30T22:32:33.670Z"}]:Array
     
           JSON.parse(response.body).with_indifferent_access